test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc

```diff
@@ -0,0 +1,219 @@
+  }
+
+  void waitForExtAuthzRequest(const std::string& expected_check_request_yaml) {
+    AssertionResult result =
+        fake_upstreams_.back()->waitForHttpConnection(*dispatcher_, fake_ext_authz_connection_);
+    RELEASE_ASSERT(result, result.message());
+    result = fake_ext_authz_connection_->waitForNewStream(*dispatcher_, ext_authz_request_);
+    RELEASE_ASSERT(result, result.message());
+
+    // Check for the validity of the received CheckRequest.
+    envoy::service::auth::v2::CheckRequest check_request;
+    result = ext_authz_request_->waitForGrpcMessage(*dispatcher_, check_request);
+    RELEASE_ASSERT(result, result.message());
+
+    EXPECT_EQ("POST", ext_authz_request_->headers().Method()->value().getStringView());
+    EXPECT_EQ("/envoy.service.auth.v2.Authorization/Check",
+              ext_authz_request_->headers().Path()->value().getStringView());
+    EXPECT_EQ("application/grpc",
+              ext_authz_request_->headers().ContentType()->value().getStringView());
+
+    envoy::service::auth::v2::CheckRequest expected_check_request;
+    TestUtility::loadFromYaml(expected_check_request_yaml, expected_check_request);
+
+    auto* attributes = check_request.mutable_attributes();
+    auto* http_request = attributes->mutable_request()->mutable_http();
+
+    // Clear fields which are not relevant.
+    attributes->clear_source();
+    attributes->clear_destination();
+    attributes->clear_metadata_context();
+    http_request->clear_id();
+    http_request->clear_headers();
+    http_request->clear_scheme();
+
+    EXPECT_EQ(check_request.DebugString(), expected_check_request.DebugString());
+
+    result = ext_authz_request_->waitForEndStream(*dispatcher_);
+    RELEASE_ASSERT(result, result.message());
+  }
+
+  void waitForSuccessfulUpstreamResponse() {
+    AssertionResult result =
+        fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_);
+    RELEASE_ASSERT(result, result.message());
+    result = fake_upstream_connection_->waitForNewStream(*dispatcher_, upstream_request_);
+    RELEASE_ASSERT(result, result.message());
+    result = upstream_request_->waitForEndStream(*dispatcher_);
+    RELEASE_ASSERT(result, result.message());
+
+    upstream_request_->encodeHeaders(Http::TestHeaderMapImpl{{":status", "200"}}, false);
+    upstream_request_->encodeData(response_size_, true);
+    response_->waitForEndStream();
+
+    EXPECT_TRUE(upstream_request_->complete());
+    EXPECT_EQ(request_body_.length(), upstream_request_->bodyLength());
+
+    EXPECT_TRUE(response_->complete());
+    EXPECT_EQ("200", response_->headers().Status()->value().getStringView());
+    EXPECT_EQ(response_size_, response_->body().size());
+  }
+
+  void sendExtAuthzResponse() {
+    ext_authz_request_->startGrpcStream();
+    envoy::service::auth::v2::CheckResponse check_response;
+    check_response.mutable_status()->set_code(Grpc::Status::WellKnownGrpcStatus::Ok);
```

Comment:
Actually, this doesn't compile, but it should be an easy fix.

Reply:
Just replace `WellKnownGrpcStatus` with `GrpcStatus`.

Reply:
Hum, so do you mean every `set_code` should be set by `Grpc::Status::` instead of value from `Grpc::Status::WellKnownGrpcStatus::` the enum? Makes sense.

Reply:
`Grpc::Status::` was renamed to `Grpc::Status::WellKnownGrpcStatus::` recently, so we need to undo that change when backporting.

Reply:
👍